File: modules/Googling.py
from googlesearch import search
from insides.bcolors import bcolors


def Googling(mail,_verbose=None):
	if _verbose != None:
		try:
			print(f"{bcolors.WARNING} -- Google Searching... [Pastebin & Throwbin]{bcolors.ENDC}")
			print(f"{bcolors.FAIL}!{bcolors.ENDC}"+"Google Search may not work properly.\n")
			urls = search(f'site:throwbin.io intext:"{mail}"') + search(f'site:pastebin.com intext:"{mail}"')
			if len(urls):
				for x in urls: 
					print(f"{bcolors.OKGREEN}|-- {bcolors.ENDC}"+x)
			else:
				print(f"{bcolors.FAIL}No Google Search result found!{bcolors.ENDC}")
		except:
			print(f"{bcolors.FAIL}Google Search error!{bcolors.ENDC}")

File: modules/Googling.go
// mosint v2.1
// Author: Alp Keskin
// Github: github.com/alpkeskin
// Website: https://imalp.co
package modules

import (
	googlesearch "github.com/rocketlaunchr/google-search"
)

func BinSearch(email string) []string {
	var url_array []string
	q := "intext:" + string('"') + email + string('"') + " site:pastebin.com"
	res, _ := googlesearch.Search(nil, q)
	size := len(res)
	for i := 0; i < size; i++ {
		url_array = append(url_array, res[i].URL)
	}
	q = "intext:" + string('"') + email + string('"') + " site:throwbin.io"
	res, _ = googlesearch.Search(nil, q)
	size2 := len(res)
	for i := 0; i < size2; i++ {
		url_array = append(url_array, res[i].URL)
	}
	return url_array
}

func Related_domains_from_google(email string) []string {
	var url_array []string
	res, _ := googlesearch.Search(nil, email)
	size := len(res)
	for i := 0; i < size; i++ {
		url_array = append(url_array, res[i].URL)
	}
	return url_array
}
